File: Chapter 02/LinearSearch.c
#include <stdio.h>
#include <stdlib.h>

int linearSearch(int *array,int arrayLength, int wantedElement)
{
    int i, j;

    for(j=0; j<arrayLength; j++) {
        if(array[j] == wantedElement) {
            return j;
        }
    }

    return -1;
}

int main()
{
    int array[10] = {1,5,11,95,36,32,15,2,9,56}, wantedElement, returnValue;

    printf("Enter the element you're looking for: ");
    scanf("%d", &wantedElement);

    returnValue = linearSearch(array,10,wantedElement);
    if(returnValue == -1) {
        printf("The element has not found.");
    }
    else {
        printf("The element has found in the %d. indictor.", returnValue);
    }

    return 0;
}

File: Chapter 02/LinearSearch.java
package linearsearch;

public class LinearSearch {
    public static void main(String[] args) {
        int[] A = {1, 2, 3, 4, 5};
        int v = 1;
        
        if(linearSearch(A, v) != -1) {
            System.out.println("found at " + linearSearch(A, v)+ " index");
        } else {
            System.out.println("not found");
        }
    }
    
    public static int linearSearch(int[]A, int v) {
        int i = -1, j;
        
        for(j=0; j<A.length; j++) {
            if(A[j] == v) {
                i = j;
                return i;
            }
        }
        
        return i;
    }
}
